lib/helpers.js

```diff
@@ -11,64 +11,25 @@ var logger = require('./logger'),
 
 exports.unzipFile = function (zipPath, cb) {
   logger.debug("Unzipping " + zipPath);
-  exports.testZipArchive(zipPath, function (err, valid) {
-    if (valid) {
-      if (exports.isWindows()) {
-        var zip = new AdmZip(zipPath);
-        zip.extractAllTo(path.dirname(zipPath), true);
-        logger.debug("Unzip successful");
-        cb(null, null);
-      } else {
-        var execOpts = {cwd: path.dirname(zipPath), maxBuffer: 524288};
-        exec('unzip -o ' + zipPath, execOpts, function (err, stderr, stdout) {
-          if (!err) {
-            logger.debug("Unzip successful");
-            cb(null, stderr);
-          } else {
-            logger.error("Unzip threw error " + err);
-            logger.error("Stderr: " + stderr);
-            logger.error("Stdout: " + stdout);
-            cb("Archive could not be unzipped, check appium logs.", null);
-          }
-        });
-      }
+    if (exports.isWindows()) {
+      var zip = new AdmZip(zipPath);
+      zip.extractAllTo(path.dirname(zipPath), true);
+      logger.debug("Unzip successful");
+      cb(null, null);
     } else {
-      cb(err, null);
-    }
-  });
-};
-
-exports.testZipArchive = function (zipPath, cb) {
-  logger.debug("Testing zip archive: " + zipPath);
-  if (exports.isWindows()) {
-    if (fs.existsSync(zipPath)) {
-      logger.debug("Zip archive tested clean");
-      cb(null, true);
-    } else {
-      cb("Zip archive was not found.", false);
-    }
-  } else {
-    var execOpts = {cwd: path.dirname(zipPath)};
-    exec("unzip -tq " + zipPath, execOpts, function (err, stderr, stdout) {
-      if (!err) {
-        if (/No errors detected/.exec(stderr)) {
-          logger.debug("Zip archive tested clean");
-          cb(null, true);
+      var execOpts = {cwd: path.dirname(zipPath), maxBuffer: 524288};
+      exec('unzip -o ' + zipPath, execOpts, function (err, stderr, stdout) {
+        if (!err) {
+          logger.debug("Unzip successful");
+          cb(null, stderr);
         } else {
-          logger.error("Zip file " + zipPath + " was not valid");
+          logger.error("Unzip threw error " + err);
           logger.error("Stderr: " + stderr);
           logger.error("Stdout: " + stdout);
-          cb("Zip archive did not test successfully, check appium server logs " +
-             "for output", false);
+          cb("Archive could not be unzipped, check appium logs.", null);
```

Comment:
can we take the opportunity to fix this? we should be calling back with an error object, e.g., `cb(new Error("Archive could not be unzipped, check appium logs."))`, and we don't want to pass another param if we have an error.